### PR TITLE
fixed some compilation errors and test failures and added testcases APIMANAGER3026DescriptionFieldTestCase, APIMANAGER3277SSOLogin

### PR DESCRIPTION
**Pull request by laki88**

### CODE REVIEW

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3277SSOLogin.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	@Test(groups = "wso2.am")
+	public void loginToPublisher() throws Exception {
+		WebDriverWait wait = new WebDriverWait(driver, 10);
+		// login to publisher
+		driver.get(publisherURL + "/site/pages/login.jag");
+		// wait until load the page
+		wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("username")));
+		driver.findElement(By.id("username")).clear();
```
Comment (prabathabey) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3277SSOLogin.java]: Maybe it's good to use constants upon regularly used fields like "username" and "password"

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/user/mgt/LoginLogoutClient.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        this.automationContext = context;
+        this.loginClient = new AuthenticatorClient(backendURL);
+    }
+
+    /**
+     * Log in to a Carbon server
+     *
+     * @return The session cookie on successful login
+     */
+    public String login()
```
Comment (imesh) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/user/mgt/LoginLogoutClient.java]: May be header comment was not auto-generated? Missing exceptions.
Thanks

File: modules/integration/tests-ui-integration/src/test/resources/testng-server-mgt.xml [hunk 3]
```diff
@@ -12,7 +12,8 @@
 
     <test name="as-integration-tests-server-mgt" preserve-order="true" parallel="false">
         <classes>
-<class name="org.wso2.am.integration.ui.tests.APIMANAGER2604ApplicationPaginationTestCase"/>
+            <class name="org.wso2.am.integration.ui.tests.APIMANAGER3026DescriptionFieldTestCase"/>
+            <class name="org.wso2.am.integration.ui.tests.APIMANAGER2604ApplicationPaginationTestCase"/>
             <!--<class name="org.wso2.am.integration.ui.tests.APIMANAGER3154StoreSearchResultDisplay"/>-->
             <!--<class name="org.wso2.am.integration.ui.tests.APIStoreLogoutTestCase"/>-->
             <!--<class name="org.wso2.am.integration.ui.tests.APIVersoinStatTestCase"/>-->
```
Comment (imesh) [inline modules/integration/tests-ui-integration/src/test/resources/testng-server-mgt.xml]: Any reason for having commented test classes?

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER2604ApplicationPaginationTestCase.java [hunk 4]
```diff
@@ -48,7 +48,7 @@ public void setUp() throws Exception {
 
     @Test(groups = "wso2.am", description = "verify pagination in application creation page")
     public void appCreationPaginationTest() throws Exception {
-        driver.get(getStoreURL());
+        driver.get(getStoreURL() + "/?tenant=carbon.super");
```
Comment (lafernando) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER2604ApplicationPaginationTestCase.java]: There is a constant for super tenant domain name in MultitenantConstants class.

File: modules/integration/tests-ui-integration/src/test/resources/testng-server-mgt.xml (hunk 3 above)
Comment (laki88) [inline modules/integration/tests-ui-integration/src/test/resources/testng-server-mgt.xml]: it is a formatting issue. I added it in line 16

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/user/mgt/LoginLogoutClient.java (hunk 2 above)
Comment (laki88) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/user/mgt/LoginLogoutClient.java]: @imesh : corrected it.

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3277SSOLogin.java (hunk 1 above)
Comment (laki88) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3277SSOLogin.java]: @prabathabey : corrected it.

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER2604ApplicationPaginationTestCase.java (hunk 4 above)
Comment (laki88) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER2604ApplicationPaginationTestCase.java]: @lafernando : Corrected it.

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/user/mgt/LoginLogoutClient.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    private static final Log log = LogFactory.getLog(LoginLogoutClient.class);
+    private String backendURL;
+    private AutomationContext automationContext;
+    private AuthenticatorClient loginClient;
+
+    public LoginLogoutClient(AutomationContext context) throws MalformedURLException,
+            XPathExpressionException, AxisFault {
+        URL backend = new URL(context.getContextUrls().getBackEndUrl());
+        backendURL = context.getContextUrls().getBackEndUrl();
+        backend.getPort();
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/user/mgt/LoginLogoutClient.java]: values retrieve though test port and get host never used it seems.

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3026DescriptionFieldTestCase.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ * The issue in APIMANAGER3026 is when api is created by rest call via publisher APIs
+ * and when we try to edit description of created API, the description was not found.
+ * This was fixed and will test by this test case.
+ */
+public class APIMANAGER3026DescriptionFieldTestCase extends AMIntegrationUiTestBase {
+
+	private WebDriver driver;
+	private APIPublisherRestClient apiPublisher;
+	private String userName = "admin";
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3026DescriptionFieldTestCase.java]: hard coding user names and passwords not recommended. you can read values though automation context

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3026DescriptionFieldTestCase.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+		String APIVersion = "1.0.0";
+
+		apiPublisher.login(userName, password);
+
+		APIRequest apiRequest = new APIRequest(APIName, APIContext, new URL(url));
+		apiRequest.setTags(tags);
+		apiRequest.setDescription(description);
+		apiRequest.setVersion(APIVersion);
+		apiRequest.setSandbox(url);
+		apiPublisher.addAPI(apiRequest);
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3026DescriptionFieldTestCase.java]: Don't we need to verify whether API is added successfully ? may be an assert

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3026DescriptionFieldTestCase.java [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+		driver.get(getPublisherURL() + "/info?name=" + APIName + "&version=" + APIVersion +
+		           "&provider=" + userName);
+
+		driver.get(getPublisherURL() + "/design?name=" + APIName + "&version=" + APIVersion +
+		           "&provider=" + userName);
+
+		WebElement descriptionElement = driver.findElement(By.id("description"));
+		String elementVal = descriptionElement.getText();
+		log.info("description value : " + elementVal);
+		Assert.assertEquals(description, elementVal);
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3026DescriptionFieldTestCase.java]: You can use testng static assert import and get rid of Assert.

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3277SSOLogin.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import org.wso2.carbon.automation.extensions.selenium.BrowserManager;
+
+/*
+ * This test should run with the following configuration
+ * Configure AM and IS to use in SSO mode
+ */
+public class APIMANAGER3277SSOLogin extends AMIntegrationUiTestBase {
+
+	private WebDriver driver;
+	private String TEST_DATA_PASSWORD = "admin", TEST_DATA_FULL_USERNAME = "admin@carbon.super";
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3277SSOLogin.java]: You should use char array to hold password even in test codes

**Pull request merged**